clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -148,6 +154,10 @@ public FileRecords slice(int position, int size) throws IOException {
      * @return the number of bytes written to the underlying file
      */
     public int append(MemoryRecords records) throws IOException {
+        if (size.get() > Integer.MAX_VALUE - records.sizeInBytes())
```

Comment:
Should we also make `size` a `volatile int` at the same time? The implementation is kind of awkward because we do two `get()`s now in this method but I think the underlying assumption is that both will return the same value. The contract would be cleaner if we document that this class is not thread safe and make sure all usages are protected by `Log` or `LogSegment` lock.

Reply:
Yeah, it's tied a bit too closely with the locking assumptions inside `Log`. For now I'll change this to use a local variable and add a note about locking somewhere.